sdk/storage/azure-storage-common/src/test/java/com/azure/storage/common/implementation/StorageCrc64CalculatorTests.java

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.Random;
+import java.util.stream.IntStream;
+import java.math.BigInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class StorageCrc64CalculatorTests {
+    private static final int KB = 1024;
+    private static final String HEX_DATA = "C8E11B40D793D1526018"; // '\\xC8\\xE1\\x1B\\x40\\xD7\\x93\\xD1\\x52\\x60\\x18'
+    private static final byte[] HEX_BYTES = hexStringToByteArray();
+
+    @ParameterizedTest
+    @CsvSource({
```

Comment:
Instead of using `CsvSource` here, I'm wondering if we should use `@MethodSource` so that we could consolidate these test cases and be more expressive in terms of the types that we use? For example, it would be interesting if we could represent the `data` argument as a byte array and then make the initial crc and expected crc longs. I think this will also help streamline any future cases we add.